test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,1828 @@
+        value = (os.path.join("core_file_bt", "misc"), None)
+        self.core_info[self.CORE_MISC] = value
+
+        # System related information
+        value = (os.path.join("system", "os_info"), None)
+        self.sys_info[self.OS_INFO] = value
+        value = (os.path.join("system", "process_info"), ["ps", "aux"])
+        self.sys_info[self.PROCESS_INFO] = value
+        value = (os.path.join("system", "etc_hosts"),
+                 ["cat", os.path.join("/", "etc", "hosts")])
+        self.sys_info[self.ETC_HOSTS] = value
+        value = (os.path.join("system", "etc_nsswitch_conf"),
+                 ["cat", os.path.join("/", "etc", "nsswitch.conf")])
+        self.sys_info[self.ETC_NSSWITCH_CONF] = value
+        value = (os.path.join("system", "lsof_pbs.out"), ["lsof"])
+        self.sys_info[self.LSOF_PBS_OUT] = value
+        value = (os.path.join("system", "vmstat.out"), ["vmstat"])
+        self.sys_info[self.VMSTAT_OUT] = value
+        value = (os.path.join("system", "df_h.out"), ["df", "-h"])
+        self.sys_info[self.DF_H_OUT] = value
+        value = (os.path.join("system", "dmesg.out"), ["dmesg"])
+        self.sys_info[self.DMESG_OUT] = value
+
+        # Miscellaneous information
+        self.misc_info[self.CTIME] = ("ctime", None)
+        self.misc_info[self.PBS_SNAPSHOT_LOG] = (self.log_filename, None)
+
+    def __initialize_snapshot(self):
+        """
+        Create a snapshot directory along with the directory structure
+        Also create a tarfile and add the snapshot dir if create_tar is True
+
+        :param snapdir: path to the snapshot directory to create
+        :type snapdir: string
+        """
+        if self.create_tar is True:
+            if os.path.isfile(self.outtar_path):
+                answer = raw_input("Target tar file by the same name " +
+                                   "already exists, to proceed please " +
+                                   "confirm overwrite " + self.outtar_path +
+                                   "? (y/n) ")
+                if answer == "y":
+                    os.remove(self.outtar_path)
+                else:
+                    raise UserQuit
+        else:
+            if os.path.isdir(self.outdir):
+                answer = raw_input("Target directory exists,  " +
+                                   "to proceed please confirm overwrite " +
+                                   self.outdir + "? (y/n) ")
+                if answer == "y":
+                    shutil.rmtree(self.outdir)
+                else:
+                    raise UserQuit
+
+        os.mkdir(self.outdir)
+
+        if self.create_tar is True:
+            self.outtar_fd = tarfile.open(self.outtar_path, "w:gz")
+
+        # Create the following directories: server, job, node, comm, queue,
+        # hook, scheduler, reservation, resource, datastore, pbs, core_file_bt
+        # and system
+        dirs_in_snapshot = ['server', 'job', 'node', 'comm', 'queue', 'hook',
```

Comment:
Since you are making use of os.makedirs() you can give full path to sub directory and parent will be created automatically